Sources/SwiftProtobuf/SwiftProtobufContiguousBytes.swift

```diff
@@ -0,0 +1,28 @@
+// Sources/SwiftProtobuf/SwiftProtobufContiguousBytes.swift
+//
+// Copyright (c) 2022 Apple Inc. and the project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See LICENSE.txt for license information:
+// https://github.com/apple/swift-protobuf/blob/main/LICENSE.txt
+//
+// -----------------------------------------------------------------------------
+
+/// Conformance to this protocol gives users a way to provide their own "bag of bytes" types
+/// to be used for serialization and deserialization of protobufs.
+///
+/// By conforming your own types to this protocol, you will be able to pass instances of said types
+/// directly to `SwiftProtobuf.Message`'s deserialisation methods
+/// (i.e. `init(serializedData:)` for binary format and `init(jsonUTF8Data:)` for JSON).
+// TODO: extend doc to include how this will be used in serialization once that API change has been finalised.
+public protocol SwiftProtobufContiguousBytes {
```

Comment:
That would basically be explaining why we want to remove `Foundation` (since `ContiguousBytes` also lives in `Foundation`). Something like:
>/// This protocol exists because the `ContiguousBytes` protocol, which would have been a viable alternative,
>/// lives inside `Foundation`.
>/// Importing `Foundation` results in very big binaries, which can be a problem for some swift-protobuf usecases.

Do we want to go there in this comment though?

Reply:
...provides a general interface for bytes since the Swift Standard Library currently does not provide such a protocol.

Or something like that?

Reply:
Yeah I think I like that better, I'll change it. Thanks.